### PR TITLE
Form Structuring

### PR DESCRIPTION
**Pull request by martinnajjar12**
### Form Structuring
In this branch, we have dealt with the forms in rails, models, controllers, and views to handle the inputs of the user and submit them to the database.
We structured the form with a plain HTML, then with form_tag, and at the end with form_for.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to new_user_path
+    else
+      @user.errors.full_messages
+      render :edit
+    end
+  end
```
Comment (BrittanyBlake) [inline app/controllers/users_controller.rb]: - [x] Update should redirect to the `edit_user_path` 👇 
Screen Shot 2020-11-04 at 12 59 02 AM

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<h1>Hello from new.html.erb</h1>
+
+<%# <form accept-charset="UTF-8" action="/users" method="post"> %>
```
Comment (BrittanyBlake) [inline app/views/users/new.html.erb]: - [x] You should be using `users_path` here 👇 
Screen Shot 2020-11-04 at 1 03 00 AM
```ruby
<form action="users_path" method="post" accept-charset="UTF-8">
```

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  <input type="text" id="username" name="user[username]">
+  <br><br>
+  <label for="email"> Input your e-mail </label><br>
+  <input type="email" id="email" name="user[email]">
+  <br><br>
+  <label for="password"> Input your Password </label><br>
+  <input type="password" id="password" name="user[password]">
+  <br><br>
+  <input type="submit" value="submit"> %>
+<%# </form> %>
```
Comment (BrittanyBlake) [inline app/views/users/new.html.erb]: - [x] You should be using the `form_authenticity_token` method.
Screenshot from Odin 👇 
Screen Shot 2020-11-04 at 1 05 53 AM
```ruby
<input
    type="hidden"
    name="authenticity_token"
    value="<%= form_authenticity_token %>"
  />
```

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  <%# <%= label_tag(:username, 'Input your username:') %><!-- <br> -->
+  <%# <%= text_field_tag(:username) %><!-- <br><br> -->
+  <%# <%= label_tag(:email, 'Enter your email:') %><!-- <br> -->
+  <%# <%= email_field_tag(:email) %><!-- <br><br> -->
+  <%# <%= label_tag(:password, 'Enter your password:') %><!-- <br> -->
+  <%# <%= password_field_tag(:password) %><!-- <br><br> -->
+  <%# <%= submit_tag('Submit') %>
+<%# <% end %>
+
+<%= form_for :user, url: users_path do |f| %>
```
Comment (BrittanyBlake) [inline app/views/users/new.html.erb]: - [x] You should be using the `@user` here from your controller.
Screenshot of requirements 👇 
Screen Shot 2020-11-04 at 1 08 51 AM
```ruby 
<%= form_for @user do |f|%> 
```

File: app/views/users/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+      <li>Your <%= e %> is invalid. Please try something else</li>
+    <% end %>
+  </ul>
+<% end %>
+<%= form_for @user, url: user_path(@user) do |f| %>
+  Username:  <%= f.text_field :username, placeholder: 'Desired username' %><br><br>
+  E-mail: <%= f.email_field :email, placeholder: 'email@company.com' %><br><br>
+  Password: <%= f.password_field :password %><br><br>
+  <%= f.submit %>
+<% end %>
```
Comment (BrittanyBlake) [inline app/views/users/edit.html.erb]: - [x] You are missing the commented out `HTML form` and `#form_tag forms` in the edit view. 
Screenshot from Odin 👇 
Screen Shot 2020-11-04 at 1 13 17 AM

**Review by BrittanyBlake — CHANGES_REQUESTED** [review]
### Status: Changes Required ♻️ 
****
### Hi Team! 👋 
You have also done a wonderful job with your project so far! 👏 👏 👏  Everything works as it shoud! 🏆 There are just a few issues that need to be resolved before you can move on. Please check the comments below for more details.⬇️ 

****
### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

File: app/views/users/new.html.erb (hunk 3 above)
Comment (sergioobolevich) [reply app/views/users/new.html.erb]: We did use the authenticity token in our form, in the first line inside the form!

Comment (BrittanyBlake) [reply app/views/users/new.html.erb]: Sorry! I must have missed that 🙈

**Review by rOluochKe — APPROVED** [review]
Hello Team
### You got it right, well done :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**